WordPress/src/main/java/org/wordpress/android/ui/stats/StatsSinglePostDetailsActivity.java

```diff
@@ -0,0 +1,711 @@
+        super.onSaveInstanceState(outState);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        if (mRestResponseParsed == null) {
+            setupEmptyUI(true);
+            mHandler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    refreshStats();
+                }
+            }, 75L);
+        } else {
+            updateUI();
+        }
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    private void refreshStats() {
+        final String remotePostID = mRemotePostItem.getItemID();
+        if (mIsUpdatingStats) {
+            AppLog.w(AppLog.T.STATS, "stats details are already updating for the following postID "
+                    + mRemotePostItem.getItemID() + ", refresh cancelled.");
+            return;
+        }
+
+        if (!NetworkUtils.isNetworkAvailable(this)) {
+            mSwipeToRefreshHelper.setRefreshing(false);
+            ToastUtils.showToast(this, this.getString(R.string.connection_error), ToastUtils.Duration.LONG);
+            return;
+        }
+
+        final RestClientUtils restClientUtils = WordPress.getRestClientUtilsV1_1();
+        final String blogId = mRemotePostItem.getBlogID();
+
+        // View and visitor counts for a site
+        final String singlePostRestPath = String.format(
+                "/sites/%s/stats/post/%s", blogId, remotePostID);
+
+        AppLog.d(AppLog.T.STATS, "Enqueuing the following Stats request " + singlePostRestPath);
+
+        RestBatchCallListener vListener = new RestBatchCallListener(this);
+        restClientUtils.get(singlePostRestPath, vListener, vListener);
+
+        mIsUpdatingStats = true;
+        mSwipeToRefreshHelper.setRefreshing(true);
+
+        mMonthsAndYearsList.setVisibility(View.GONE);
+        mAveragesList.setVisibility(View.GONE);
+        mRecentWeeksList.setVisibility(View.GONE);
+
+    }
+
+    private void setupEmptyUI(boolean isLoading) {
+        Context context = mGraphContainer.getContext();
+        if (context != null) {
+            LayoutInflater inflater = LayoutInflater.from(context);
+            View emptyBarGraphView = inflater.inflate(R.layout.stats_bar_graph_empty, mGraphContainer, false);
+            final TextView emptyLabel = (TextView) emptyBarGraphView.findViewById(R.id.stats_bar_graph_empty_label);
+            if (isLoading) {
+                emptyLabel.setText("");
+            } else {
+                emptyLabel.setText(getString(R.string.stats_bar_graph_empty));
+            }
+            if (emptyBarGraphView != null) {
```

Comment:
emptyBarGraphView should never be null here since it's inflated above (and if it is null, the above findViewById for emptyLabel will NPE before this point).